ibis/backends/clickhouse/datatypes.py

```diff
@@ -1,179 +1,104 @@
 from __future__ import annotations
 
-import functools
-from typing import TYPE_CHECKING, Literal
+from typing import Literal
 
-import sqlglot as sg
-from sqlglot.expressions import ColumnDef, DataType
+import sqlglot.expressions as sge
 
 import ibis
 import ibis.expr.datatypes as dt
+from ibis.backends.base.sqlglot.datatypes import SqlglotType
 from ibis.common.collections import FrozenDict
-from ibis.formats.parser import TypeParser
 
-if TYPE_CHECKING:
-    from collections.abc import Mapping
-
-    from sqlglot.expressions import Expression
-
-    try:
-        from sqlglot.expressions import DataTypeParam
-    except ImportError:
-        from sqlglot.expressions import DataTypeSize as DataTypeParam
+typecode = sge.DataType.Type
 
 
+# TODO(kszucs): add a bool converter method to support different clickhouse bool types
 def _bool_type() -> Literal["Bool", "UInt8", "Int8"]:
     return getattr(getattr(ibis.options, "clickhouse", None), "bool_type", "Bool")
 
 
-class ClickHouseTypeParser(TypeParser):
-    __slots__ = ()
-
+class ClickhouseType(SqlglotType):
     dialect = "clickhouse"
     default_decimal_precision = None
     default_decimal_scale = None
     default_nullable = False
 
-    short_circuit: Mapping[str, dt.DataType] = FrozenDict(
+    unknown_type_strings = FrozenDict(
         {
-            "IPv4": dt.INET(nullable=default_nullable),
-            "IPv6": dt.INET(nullable=default_nullable),
-            "Object('json')": dt.JSON(nullable=default_nullable),
-            "Array(Null)": dt.Array(dt.null, nullable=default_nullable),
-            "Array(Nothing)": dt.Array(dt.null, nullable=default_nullable),
+            "ipv4": dt.INET(nullable=default_nullable),
+            "ipv6": dt.INET(nullable=default_nullable),
+            "object('json')": dt.JSON(nullable=default_nullable),
+            "array(null)": dt.Array(dt.null, nullable=default_nullable),
+            "array(nothing)": dt.Array(dt.null, nullable=default_nullable),
```

Comment:
These types are case sensitive, why did you alter their case?

Reply:
From_string converts the input string to lowercase before looking up the types unknown to sqlglot